compiler/cpp/tests/ocaml/snapshot_typedefs.cc

```diff
@@ -0,0 +1,5 @@
+const char* snapshot(R"""(
```

Comment:
missing ASF header (in most added files actually)

Reply:
Sure, added header and squashed into a single commit.